src/sage/knots/link.py

```diff
@@ -1275,6 +1341,14 @@ def khovanov_homology(self, ring=ZZ, height=None, degree=None):
             sage: L.khovanov_homology(degree=1, height=1)
             {}
         """
+        from sage.interfaces.khoca import check_kwds
+        check_kwds(**kwds)
+        khoca = False
+        if implementation == 'Khoca':
+            khoca = True
+        elif implementation != 'native':
+            raise ValueError('%s is not a recognized implementation')
```

Comment:
```suggestion
        khoca = False
        if implementation == 'Khoca':
            khoca = True
            from sage.interfaces.khoca import check_kwds
            check_kwds(**kwds)
        elif implementation != 'native':
            raise ValueError('%s is not a recognized implementation')
```
There might be a mismatch of the keywords if not using Khoca (in the future).

Reply:
> There might be a mismatch of the keywords if not using Khoca (in the future).

Sorry, I forgot what I was intending here. I wanted to avoid things like in the following doctest which is now [failing](https://github.com/sagemath/sage/actions/runs/17061754161/job/48369831551?pr=40081#step:11:1002) in the current [CI-run](https://github.com/sagemath/sage/actions/runs/17061754161/job/48369831551)


```
sage -t --long --warn-long 30.0 --random-seed=286735480429121101562228604801325644303 src/sage/knots/link.py
**********************************************************************
Error: Failed example:: Got: {0: {0: Z}, 2: {0: Z, 1: 0, 2: 0}, 4: {0: 0, 1: 0, 2: Z}, 6: {2: Z}}


    K.khovanov_homology(base_ring=QQ)
Expected:
    Traceback (most recent call last):
    ...
    ValueError: 'base_ring' is not a valid KnownKeywords
Got:
    {0: {0: Z}, 2: {0: Z, 1: 0, 2: 0}, 4: {0: 0, 1: 0, 2: Z}, 6: {2: Z}}
**********************************************************************
1 item had failures:
   1 of  19 in sage.knots.link.Link.khovanov_homology
    [596 tests, 1 failure, 6.11s
```

Do you have another suggestion or shall I just revert this?

Reply:
```python
elif kwds:
    raise ValueError(f"invalid keyword(s): {list(kwds)}")
```

Reply:
Although you could be more specific when testing the generic implementation. I think it is also okay to ignore unused keywords, but it probably is good to document that though.

Reply:
> Although you could be more specific when testing the generic implementation. I think it is also okay to ignore unused keywords, but it probably is good to document that though.

But, ignoring would change the existing behavior, as the user would no longer notice a typo in one of the existing keywords (i.e., `khovanov_homology(rimg=QQ)` would return the homology over `ZZ`). Therefore I just do your suggestion above in https://github.com/sagemath/sage/commit/240e7ce3056c3d9e9ad2f4140db4d034b8a84844.

Reply:
I think of it as a form of "garbage-in, garbage-out," where we don't have to protect the user from every possible mistake (focusing on the ones that break the code in strange ways). It's more of a coincidence that it would fail before, as the function signature was different. Anyways, I am okay with raising the error too. Thanks.